### PR TITLE
Request and receive block headers

### PR DESCRIPTION
**Pull request by jomuel**

### CODE REVIEW

File: src/block_manager/block_manager.rs [hunk 1]
```diff
@@ -1,23 +1,60 @@
...
+    common_types::{BlockId, GetBlockHeaders},
+    scheduler::PeerOrganizer,
+    scheduler::peer_organizer::{ErrorAct, PeerId, Task},
+    scheduler::protocol::{ProtocolId, MessageId, EthMessageId}
+};
 
 pub struct BlockManager {
-    // chain: Arc<dyn Blockchain>,
+    chain: Arc<Mutex<dyn Blockchain + Send + Sync>>,
+    peer_organizer: Arc<Mutex<PeerOrganizer>>
```
Comment (rakita) [inline src/block_manager/block_manager.rs:21]: I don't like this coupling, is it possible for BlockManger to be a standalone component without peer_organizer?

File: src/scheduler/peer_organizer.rs [hunk 2]
```diff
@@ -34,10 +34,12 @@ pub enum Task {
     InsertPeer(HandshakeInfo),
     PenalPeer(PeerId, PeerPenal, String), //last is reason
     WaitForStatus(PeerId, MessageData),
+    GetBlocks(PeerId, MessageData),
```
Comment (rakita) [inline src/scheduler/peer_organizer.rs:37]: GetBlockHeaders?

File: src/block_manager/block_manager.rs [hunk 3]
```diff
@@ -1,23 +1,60 @@
...
+        Arc::new(Mutex::new(BlockManager { chain, peer_organizer }))
+    }
+
+    fn request_block_headers(&self, peer: &PeerId) -> Task {
+        let request = GetBlockHeaders::new(BlockId::Number(10_000_000), 100, 0, false);
+        let data = encode_get_block_headers(&request);
+        Task::GetBlocks(*peer, data)
+    }
+
+    pub fn sync(&self) {
```
Comment (rakita) [inline src/block_manager/block_manager.rs:41]: Hm this is probably why you needed peer_organizer. My thoughts on this are, you can create a Task without peer_id and send that to peer_organizer to be scheduled and inside of `schedule` function, you choose a random peer that will receive this task.

**Review by rakita — APPROVED** [review]
lgtm, you can merge this and improve it in next iteration.

File: src/scheduler/peer_organizer.rs (hunk 2 above)
Comment (jomuel) [reply src/scheduler/peer_organizer.rs:37]: I started implementing the `(Get)BlockBodies` requests yesterday, and already changed the name of the enum variant to `InitialRequest` (i. e. not a response to a peer request) with a  message id parameter.

File: src/block_manager/block_manager.rs (hunk 1 above)
Comment (jomuel) [reply src/block_manager/block_manager.rs:21]: Should work as you suggested below, I will include it in the next PR.

**Pull request merged**